providers/shared/references/Certificate/id.ftl

```diff
@@ -54,7 +54,12 @@
 [#function getCertificatePrimaryDomain certificateObject]
     [#list certificateObject.Domains as domain]
         [#if isPrimaryDomain(domain) ]
-            [#return domain ]
+            [#if !domain.Name?contains(".") ]
+                [@fatal message="Missing or invalid primary domain name" context=domain /]
+                [#return {} ]
+            [#else]
+                [#return domain ]
+            [/#if]
             [#break]
         [/#if]
     [/#list]
```

Comment:
Again I think this isn't quite the right spot to perform this validation. 
This function is specifically for validating a primary domain, What about the other domains that are generated in the certificateObject? 

Probably best to do this validation when the certificates object is created. That way it applies whenever a domain name is formed.

Reply:
I think this is now in the correct place